api/tests/unit/domain/usecases/unarchive-campaign_test.js

```diff
@@ -1,50 +1,25 @@
-const { expect, sinon, catchErr } = require('../../../test-helper');
+const { expect, sinon } = require('../../../test-helper');
 const unarchiveCampaign = require('../../../../lib/domain/usecases/unarchive-campaign');
-const { UserNotAuthorizedToUpdateCampaignError } = require('../../../../lib/domain/errors');
+const Campaign = require('../../../../lib/domain/models/CampaignForArchiving');
 
 describe('Unit | UseCase | unarchive-campaign', function () {
-  const userId = 'user id';
-  const campaignId = 'campaign id';
-  let campaignRepository;
+  let campaignForArchivingRepository;
 
   beforeEach(function () {
-    campaignRepository = {
-      update: sinon.stub(),
-      checkIfUserOrganizationHasAccessToCampaign: sinon.stub(),
+    campaignForArchivingRepository = {
+      get: sinon.stub(),
+      save: sinon.stub(),
     };
   });
 
-  context('when the user has the rights to update the campaign', function () {
-    beforeEach(function () {
-      campaignRepository.checkIfUserOrganizationHasAccessToCampaign.withArgs(campaignId, userId).resolves(true);
-      return unarchiveCampaign({ campaignId, userId, campaignRepository });
-    });
+  it('archives the campaign', async function () {
```

Comment:
**suggestion** : `unarchives the campaign`

Reply:
done